### ISSUE TITLE
Wrong configuration for Nginx reverse proxy on zeronet.io

### ISSUE THREAD

**Issue by nomagics** [HelloZeroNet/ZeroNet#2542]
Hello,

Always need to open new issues because the maniac Tamas Kocsis closing issues without any useful contribution to solve those issues. He not even reading them. 

The problem is that on zeronet.io (https://zeronet.io/docs/faq/#how-to-configure-nginx-reverse-proxy-for-zeronet) which linked by @HelloZeroNet here https://github.com/HelloZeroNet/ZeroNet/issues/2541#issuecomment-619966957 is wrong! 

If someone want to set-up a reverse proxy Proxypass should point to 127.0.0.1:43110/**RAW**/bitcoinaddress or the search engines will not able to index the site, if by some magic some robot do index it than the **entire website will be penalized**!

Frames are not for search engines so stop recommending bullshit to people! Also as I said here https://github.com/HelloZeroNet/ZeroNet/issues/2541#issuecomment-620092824 the headers what you wrote to the config file you unfortunately recommend SHOULD NOT contain headers at all because in a reverse proxy the headers are sent by the front-end and not the back-end! If you send the headers with the back-end you will mess up things you never even imagined and as a programmer I can tell there is no logic behind that because you sending other headers with UiRequest.py as well. 

Providing different headers in the core, back-end and on the front-end is irresponsible and stupid. Headers should be only presented at the publicly facing front-end and as I said proxypass SHOULD point to 127.0.0.1:43110/**RAW**/bitcoinaddress because your fcking ZeroFrame are not needed and useless on the public internet!

**Comment by HelloZeroNet** [comment]
If you redirect to /raw, then ZeroNet API commands won't work, so probably it's not what you want unless you want host host only static files.

**Comment by nomagics** [comment]
How you host dynamic websites if the content is not even available for the general public because of your frame? So fucking stupid! If the config setup good you can run dynamic websites without frames because you UNSETTING HEADERS on the back-end which will pass the data to front-end over firewall. The frame only presented to the server running ZeroNet locally but you can escape the frame with reverse proxy in anyway. If you don't escape the frame **YOU WILL BE PENALIZED!!!** @filips123 @imachug Are this guy is really this retarded? Fuck you Tamas, you are disgustingly stupid.

**Comment by filips123** [comment]
Iframe wrapper is there for a reason. If there won't be one, every site will be able to access data from all other sites...

**Comment by nomagics** [comment]
I know @filips123 that is why I kept the frame in my fork. But for a reverse proxy which want to display the site on example.com don't need frame! Why would? 

The back-end is loaded in frame but the front-end is not and should not if you don't wanna be penalized.

**Comment by filips123** [comment]
You also need iframe on revere proxies because if not, sites would have direct access to WebSocket API which basically means NOSANDBOX and ADMIN permissions. Also, wrapper manages many of ZeroFrame functionalities (prompts, notifications...) so you won't get them if you remove it.

**Comment by purplesyringa** [comment]
The problem is not in security in case of reverse proxy for a single site, it's because of usability. If you really want to remove iframe, you'll have to implement the wrapper (i.e. notifications and ZeroFrame) yourself. Not an impossible task but not a simple one either.

**Comment by nomagics** [comment]
There is some misunderstanding I see. @filips123 

There is two server;

- one with zeronet and some web server on the back-end. (reverse proxy)
- other on a front-end with a web server and config for example.com

front-end over firewall connect to xxx.xxx.xxx.xxx:80 (which is the back-end)
the reverse proxy pass from port 80 to port 43110 for **RAW** or for frame and back.

The back-end is not exposed to the public at all! It is sitting behind a firewall (actually two firewalls)

Regarding the headers. 
Zeronet send headers using UiRequest.py which can be unset or changed on the back-end, if you send other headers using the web server which is the reverse proxy than you send from both UiRequest.py and from the web server MIXED headers. 

If you pass the data to front-end and the front-end has a web server (which facing publicly) than you will end up with headers supplied from 3 different place. The better is to unset headers on the web server running on back-end and add the necessary headers on the front-end.

Frames are not needed and useless on the public Internet. You will be 100% going to be penalized by search engines!

**Comment by nomagics** [comment]
> The problem is not in security in case of reverse proxy for a single site, it's because of usability. If you really want to remove iframe, you'll have to implement the wrapper (i.e. notifications and ZeroFrame) yourself. Not an impossible task but not a simple one either.

Actually I did removed even notifications from my fork. The coffee scripts are a big mess if you ask me so I just keep deleting the part I don't need. I also changed port numbers! The only thing I did not changed in the new software built on ZeroNet is the peer_port anything else is changed. 

I also implemented the UTC timestamp and not using time.time() which returns local time... I also doubled the RSA key and changed the curve to 384, blake3 hashing etc... You wanna take a look imachug?
